In [30]:
#Import library 
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

#data importing

SS=pd.read_excel('Store Sales.xlsx')

SS.head(5)

Row ID Order Priority  Discount  Unit Price  Shipping Cost  Customer ID  \
0  18606.0  Not Specified      0.01        2.88           0.50          2.0   
1  20847.0           High      0.01        2.84           0.93          3.0   
2  23086.0  Not Specified      0.03        6.68           6.15          3.0   
3  23087.0  Not Specified      0.01        5.68           3.60          3.0   
4  23088.0  Not Specified      0.00      205.99           2.50          3.0   

     Customer Name    Ship Mode Customer Segment Product Category  ...  \
0  Janice Fletcher  Regular Air        Corporate  Office Supplies  ...   
1    Bonnie Potter  Express Air        Corporate  Office Supplies  ...   
2    Bonnie Potter  Express Air        Corporate  Office Supplies  ...   
3    Bonnie Potter  Regular Air        Corporate  Office Supplies  ...   
4    Bonnie Potter  Express Air        Corporate       Technology  ...   

  State or Province       City  Postal Code Order Date Ship Date    Profit  \
0          Illinois    Addison      60101.0    41057.0   41059.0    1.3200   
1        Washington  Anacortes      98221.0    40366.0   40367.0    4.5600   
2        Washington  Anacortes      98221.0    40751.0   40752.0  -47.6400   
3        Washington  Anacortes      98221.0    40751.0   40752.0  -30.5100   
4        Washington  Anacortes      98221.0    40751.0   40751.0  998.2023   

   Quantity ordered new    Sales  Order ID     Cost  
0                   2.0     5.90   88525.0     5.76  
1                   4.0    13.01   88522.0    11.36  
2                   7.0    49.92   88523.0    46.76  
3                   7.0    41.64   88523.0    39.76  
4                   8.0  2893.34   88523.0  1647.92  

[5 rows x 24 columns]

In [31]:
SS.shape

(200, 24)

### 1. Find the Total Sales by region

In [32]:
a=SS.groupby('Region').sum()['Sales']
a

Region
Central     25883.87
East       172056.49
South        4233.98
West       119210.91
Name: Sales, dtype: float64

### 2. Find the Total sales by city

In [33]:
b=SS.groupby('City').sum()['Sales']
b

City
Addison                  5.90
Alton                  312.31
Anacortes             9985.91
Bedford                705.47
Bellevue              1678.51
Burlington            2801.06
Camden                3588.54
Colonial Heights       463.01
Coppell               1177.22
Cranston                90.39
Duxbury                526.36
Eugene                  99.56
Fall River             465.90
Grants Pass          14713.55
Great Falls             46.93
Gresham              22978.90
Helena                4041.16
Hillsboro              346.79
Iselin                 294.52
Keizer                 125.18
Laguna Hills          5113.07
Laguna Niguel         4779.64
Lake Elsinore         4040.79
Lake Forest           1716.07
Lakewood              5172.20
Lancaster              229.44
Lawndale               332.05
Missoula               463.58
Mountain View         4790.26
Murrieta              4346.75
Napa                 11320.74
New York City       130910.17
Pennsauken             413.90
Port 

### 3. Find the Total sales by state or province

In [34]:
c=SS.groupby('State or Province').sum()['Sales']
c

State or Province
California        52821.07
Illinois           7550.92
Maine               240.14
Massachusetts       992.26
Minnesota          2445.42
Montana            4551.67
Nebraska           1678.51
New Hampshire       705.47
New Jersey         6681.09
New York         156875.90
Oregon            38263.98
Rhode Island       3760.57
Texas             14209.02
Vermont            2801.06
Virginia           4233.98
Washington        23574.19
Name: Sales, dtype: float64

### 4. Find the total profit by region


In [35]:
d=SS.groupby('Region').sum()['Profit']
d

Region
Central     3792.145476
East        9239.878280
South       -222.141400
West       19246.948452
Name: Profit, dtype: float64

### 5. Find the total profit by city


In [36]:
e=SS.groupby('City').sum()['Profit']
e

City
Addison                1.320000
Alton                -22.184000
Anacortes           3242.669900
Bedford             -158.740000
Bellevue           -1025.284000
Burlington         -2426.550000
Camden              -346.615198
Colonial Heights     -56.468000
Coppell              287.374400
Cranston              23.120000
Duxbury              234.680000
Eugene               -43.331600
Fall River            72.992800
Grants Pass         3317.767800
Great Falls         -103.150000
Gresham             2249.990340
Helena               905.061600
Hillsboro             77.084000
Iselin                23.010000
Keizer                 7.670000
Laguna Hills        1580.802060
Laguna Niguel       1344.720100
Lake Elsinore       1317.296860
Lake Forest          392.157348
Lakewood            1035.933012
Lancaster             26.602800
Lawndale            -122.209440
Missoula            -114.639900
Mountain View       1312.038000
Murrieta            1547.318100
Napa                1396.506200
New

### 6. Find the total profit by state or province


In [37]:
f=SS.groupby('State or Province').sum()['Profit']
f

State or Province
California       16328.711540
Illinois          2674.438416
Maine               82.848300
Massachusetts      307.672800
Minnesota          697.160500
Montana            687.271700
Nebraska         -1025.284000
New Hampshire     -158.740000
New Jersey         309.207202
New York         10155.006978
Oregon            5609.180540
Rhode Island       970.433000
Texas             1445.830560
Vermont          -2426.550000
Virginia          -222.141400
Washington       -3378.215328
Name: Profit, dtype: float64

### 7. Find out the total sales by customer

In [38]:
g=SS.groupby('Customer Name').sum()['Sales']
g

Customer Name
Alexandra Fuller           1207.58
Alice Capps                  99.56
Amanda Hardy Singleton      240.14
Anne Nguyen                 463.01
Annette Boone              2801.06
                            ...   
Terri Lyons                3373.24
Theodore Moran             3607.01
Timothy Reese               538.72
Tony Wilkins Winters      24618.11
Virginia Hardison          4040.79
Name: Sales, Length: 62, dtype: float64

### 8. Find out Which customer segment is more profitable

In [39]:
g=SS.groupby('Customer Segment').sum()['Profit']
g

Customer Segment
Consumer           5823.679946
Corporate         16692.249676
Home Office        4912.184506
Small Business     4628.716680
Name: Profit, dtype: float64

### 9. Find out the total number of orders by order priority

In [40]:
h=SS.groupby('Order Priority').count()['Order ID']
h

Order Priority
Critical         38
High             43
Low              38
Medium           35
Not Specified    46
Name: Order ID, dtype: int64

In [41]:
h.min()

35

### 10. Find out the total number of orders by ship mode

In [42]:
k=SS.groupby('Ship Mode').count()['Order ID']
k

Ship Mode
Delivery Truck     25
Express Air        27
Regular Air       148
Name: Order ID, dtype: int64

### 11. Find out the total sales by product category


In [43]:
l=SS.groupby('Product Category').sum()['Sales']
l

Product Category
Furniture           75395.16
Office Supplies    118421.10
Technology         127568.99
Name: Sales, dtype: float64

### 12. Find out the total sales by product Sub-Category


In [44]:
la=SS.groupby('Product Sub-Category').sum()['Sales']
la

Product Sub-Category
Appliances                         9800.85
Binders and Binder Accessories    46282.84
Bookcases                          6104.75
Chairs & Chairmats                52905.09
Computer Peripherals               4260.61
Copiers and Fax                    4314.50
Envelopes                         22185.52
Labels                             1800.71
Office Furnishings                 7276.81
Office Machines                   36055.44
Paper                             10482.01
Pens & Art Supplies                2175.31
Rubber Bands                        727.69
Scissors, Rulers and Trimmers       298.05
Storage & Organization            24668.12
Tables                             9108.51
Telephones and Communication      82938.44
Name: Sales, dtype: float64

### 13. Find out the total profit by product category


In [45]:
ka=SS.groupby('Product Category').sum()['Profit']
ka

Product Category
Furniture          -3388.408623
Office Supplies    21496.538840
Technology         13948.700592
Name: Profit, dtype: float64

### 14. Find out the Total profit by product subcategory



In [46]:
na=SS.groupby('Product Sub-Category').sum()['Profit']
na

Product Sub-Category
Appliances                         3189.057460
Binders and Binder Accessories    11321.746600
Bookcases                          -387.654000
Chairs & Chairmats                -3065.547520
Computer Peripherals               1690.071400
Copiers and Fax                     345.000000
Envelopes                          3881.970600
Labels                              399.567600
Office Furnishings                 1623.870780
Office Machines                   -2765.862324
Paper                              3204.024860
Pens & Art Supplies                  27.419100
Rubber Bands                         -6.938000
Scissors, Rulers and Trimmers       -24.970000
Storage & Organization             -495.339380
Tables                            -1559.077884
Telephones and Communication      14679.491516
Name: Profit, dtype: float64

### 15. Find the total sales by order date


In [47]:
na1=SS.groupby('Order Date').sum()['Sales']
na1

Order Date
40180.0    21661.40
40206.0       69.74
40211.0    14238.76
40221.0     1910.78
40231.0      849.36
             ...   
41596.0     3183.42
41604.0      608.10
41619.0      449.66
41620.0      439.92
41635.0    19221.06
Name: Sales, Length: 119, dtype: float64

### 16. Find the total profit by order date

In [48]:
b2=SS.groupby('Order Date').sum()['Profit']
b2

Order Date
40180.0    -567.628000
40206.0     -33.604000
40211.0   -4853.100000
40221.0     655.363100
40231.0     144.690000
              ...     
41596.0     353.772160
41604.0     106.756192
41619.0     -48.236352
41620.0    -103.968480
41635.0    6468.189324
Name: Profit, Length: 119, dtype: float64

# Conclusion

This is a Store Sales Data. 

The data contains 200 rows and 24 columns

1) There are no null values in this data.

2)To find total sales by region and the highest sales in the east region.

3)To find total sales by region and the smallest sales in the South.

4)To find total profit by region and the highest profit in the west.

5)To find total profit by region and the smallest profit in the South.

6)To find total profit by customer segment and the highest sales in the corporate.

7)To find total profit by customer segment and the smallest profit in the small business.

8)Find the Total sales by city give sales of Camden-3588.54

9)Find the Total sales by state or province-52821.07

10)Find the total profit by region (South) -222.141400

11)Find the total profit by city (Iselin) -23.010000

12)What is the total profit of province montana?

The total profit of this sales in province Montana is 687.271700

13) which state have got more profit in this sales?

State have got more profit in California

14) which region profit has not reached to a good level?

region profit has not reached to a good level in South

15) Which customer segment is more profitable

customer segment is more profitable in Corporate

16) which mode of order priority least orders has been taken place

medium 35 is the answer

17)which ship mode customers are mostly buying Products

through which ship mode customers are mostly buying Products

customers are mostly buying Products in regular air 148

18)what is the total sales of paper

Total sales of paper in 10482.01

19)order priority, categorical or continues variable?

order priority, categorical variable


